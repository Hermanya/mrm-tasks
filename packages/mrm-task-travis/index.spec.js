/* eslint-disable no-console */
'use strict';

jest.mock('fs');
jest.mock('mrm-core/src/util/log', () => ({
	added: jest.fn(),
}));

const getConfigGetter = require('../../mrm').getConfigGetter;
const vol = require('memfs').vol;
const task = require('./index');

const console$log = console.log;

const json = o => JSON.stringify(o, null, '  ');

const packageJson = json({
	name: 'unicorn',
	engines: {
		node: 4,
	},
	scripts: {
		test: 'jest',
	},
});

beforeEach(() => {
	console.log = jest.fn();
});

afterEach(() => {
	vol.reset();
	console.log = console$log;
});

it('should add Travis CI', () => {
	vol.fromJSON({
		'/package.json': packageJson,
	});

	task(getConfigGetter({}));

	expect(vol.toJSON()).toMatchSnapshot();
});
