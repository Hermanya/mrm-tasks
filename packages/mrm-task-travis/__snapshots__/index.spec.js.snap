// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add Travis CI 1`] = `
Object {
  "/.travis.yml": "language: node_js
cache:
  directories:
    - node_modules
node_js:
  - 4
  - 6
  - 8
",
  "/package.json": "{
  \\"name\\": \\"unicorn\\",
  \\"engines\\": {
    \\"node\\": 4
  },
  \\"scripts\\": {
    \\"test\\": \\"jest\\"
  }
}",
}
`;

exports[`should add a badge to the Readme 1`] = `
"# Unicorn

[![Build Status](https://travis-ci.org/gh/unicorn.svg)](https://travis-ci.org/gh/unicorn)"
`;

exports[`should add latest Node version if engines field is not defined 1`] = `
Object {
  "/.travis.yml": "language: node_js
cache:
  directories:
    - node_modules
node_js:
  - 8
",
  "/package.json": "{
  \\"name\\": \\"unicorn\\",
  \\"scripts\\": {
    \\"test\\": \\"jest\\"
  }
}",
}
`;
